### PR TITLE
Use craco instead of CRA-rewired

### PR DESCRIPTION
**Pull request by bbecquet**
Switching to `craco` to replace `react-app-rewired`, as suggested in https://github.com/CartoDB/carto-react-template/pull/334#issuecomment-1082695416

For now just in `template-sample-app-3` to see how it works (spoiler, it's pretty simple :slightly_smiling_face:)

### CODE REVIEW

**Comment by VictorVelarde** [comment]
This fails in CI, but I would suggest dealing with it in this same PR: basically switching (extending?) carto-2 version with carto-3 counterparts

**Comment by bbecquet** [comment]
> This fails in CI, but I would suggest dealing with it in this same PR: basically switching (extending?) carto-2 version with carto-3 counterparts

@VictorVelarde Sorry, I'm not sure what you mean. Do you propose to write also a Cypress test-suite on v3?

It seems to me the CI currently fails on v2 because tests depend on external data:

```js
getAggregationSumWidget().findByText('411M').should('exist');
```

If `carto-demo-data.demo_tilesets.osm_buildings` has been updated since the test spec was written, this `411M` value has probably changed. IMO we should rewrite this test to check something else.

File: template-sample-app-2/template/cypress/integration/tileset_page_spec.js [hunk 1]
```diff
@@ -33,6 +33,8 @@ describe('Tileset Page', () => {
     zoomOut.click();
 
     // Total aggregated sum
-    getAggregationSumWidget().findByText('411M').should('exist');
+    getAggregationSumWidget()
+      .findByText(/^4\d\d(\.\d+)M$/)
```
Comment (bbecquet) [inline template-sample-app-2/template/cypress/integration/tileset_page_spec.js:37]: It's not pretty, but it's more robust… (at least while the number of buildings in the dataset is < 500 M)

**Comment by VictorVelarde** [comment]
Notice that today, craco is not still officially supported with cra-v5 (they are working on it for v7.0.0, currently on alpha)

File: template-sample-app-2/template/cypress/integration/tileset_page_spec.js (hunk 1 above)
Comment (VictorVelarde) [reply template-sample-app-2/template/cypress/integration/tileset_page_spec.js:37]: Ok, I'm gonna try to deal with it in another PR here: https://github.com/CartoDB/carto-react-template/pull/336, where I'm facing same issue

Comment (VictorVelarde) [reply template-sample-app-2/template/cypress/integration/tileset_page_spec.js:37]: Ok, I wasn't able to finish the CI upgrade on #336, so I moved the WIP to #337 (where locally all e2e tests pass for v3)

**Comment by VictorVelarde** [comment]
Done in #356

**Pull request closed without merging**